### ISSUE TITLE
PD TC UX: weirdness with blurred temperature field

### ISSUE THREAD

**Issue by howisthisnamenottakenyet** [Opentrons/opentrons#6141]
## Background

When you are editing an existing Thermocycler step and toggle the block or lid to activated, the empty temperature field causes an error. The temperature field has not been blurred yet. 
![image.png](https://images.zenhubusercontent.com/5e27344f9aca8806bc57bc3b/a426b60d-9e63-4e7d-90b7-fe558f48063e)

## Acceptance criteria
Thermocycler temperature fields show errors after being blurred, not when pristine

**Comment by IanLondon** [comment]
Currently, all fields are non-pristine when you re-open an existing step form. The temperature fields here are controlled by a toggle. In other forms, eg Mix/Transfer advanced settings, we have fields that are controlled by a checkbox -- but they are never validated. Eg the advanced setting mix does not produce an error even when you do `Mix ✓  0 uL     _ times` -- we don't have validation rules for it, we let you mix 0uL if you really want to, and we treat _ times as zero times. If the advanced settings fields were validated, they would have the same behavior as the temperature fields under the toggles.

TL;DR - here we should probably consider the behavior of all fields that are conditionally shown based on a toggle/checkbox, not just in TC form. We shouldn't just one-off this IMO (not sure we even could :stuck_out_tongue: ).

Maybe we can define something like `const FIELDS_CONTROLLED_BY_TOGGLE = ['mix_repetitions', 'block_temp_etc']` and use that to make the behavior "all fields are non-pristine upon opening a previously saved form" change to "all fields are non-pristine upon opening a previously saved form, unless they are controlled by a toggle and that toggle is off"

PS I'm using toggle/checkbox interchangeably, they're different widgets but doing the same thing here

---

Tricky biz: if we do this, and we don't clear out the values of all fields controlled by toggle, then we'll wind up in a bad situation. If the user types `Block temp: 9999`, unchecks the toggle, saves the form, re-opens the form, then turns on the toggle, they have "9999 C" but don't see an error about exceeding the max. (I think we were careful to clear out the values for these 4 TC temp fields, but for the advanced setting checkbox-controlled fields we definitely don't clear them out when the checkbox is toggled off)

Ideally, we can use the FIELDS_CONTROLLED_BY_TOGGLE info to clear the controlled fields out when their toggle/checkbox is turned off (in handleFormChange), instead of individually handing each field by name
